.github/workflows/integration-tests.yml

```diff
@@ -2,47 +2,84 @@ name: Integration tests
 
 on:
   workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to run the tests against'
+        type: choice
+        required: true
+        default: 'dev'
+        options:
+          - dev
+          - staging
   workflow_call:
+    inputs:
+      environment:
+        default: 'staging'
+        required: false
+        type: string
     secrets:
-      FIREBOLT_USERNAME:
-        required: true
-      FIREBOLT_PASSWORD:
-        required: true
+      FIREBOLT_USERNAME_STAGING:
+        required: false
+      FIREBOLT_PASSWORD_STAGING:
+        required: false
+      FIREBOLT_CLIENT_ID_STAGING:
+        required: false
+      FIREBOLT_CLIENT_SECRET_STAGING:
+        required: false
+      FIREBOLT_USERNAME_DEV:
+        required: false
+      FIREBOLT_PASSWORD_DEV:
+        required: false
+      FIREBOLT_CLIENT_ID_DEV:
+        required: false
+      FIREBOLT_CLIENT_SECRET_DEV:
+        required: false
 
 jobs:
   integration-tests:
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
-        uses: actions/checkout@v2
-
+        uses: actions/checkout@v3
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
           go-version: '1.19.0'
-
+      - name: Determine env variables
+        run: |
+          if [ "${{ github.event.inputs.environment }}" == 'staging' ]; then
+             echo "USERNAME=${{ secrets.FIREBOLT_USERNAME_STAGING }}" >> "$GITHUB_ENV"
+             echo "PASSWORD=${{ secrets.FIREBOLT_PASSWORD_STAGING }}" >> "$GITHUB_ENV"
+             echo "CLIENT_ID=${{ secrets.FIREBOLT_CLIENT_ID_STAGING }}" >> "$GITHUB_ENV"
+             echo "CLIENT_SECRET=${{ secrets.FIREBOLT_CLIENT_SECRET_STAGING }}" >> "$GITHUB_ENV"
```

Comment:
I'm worried this approach might leak the credentials in logs. GH replaces secrets with `***` by default, but what if this environment-agnostic `CLIENT_SECRET` gets dumped in a log?

Reply:
At the end of the day they are all stored in the environment variables (eg: if you do 'env' in those boxes you'll get the list of all variables in clear, which includes the passwords). As long as we don't decide to logs them we're good 🧷 
 Plus, those are development environments which should not contain any critical data

Reply:
Yeah, I've run `env` and it doesn't print sensitive informtaion https://github.com/firebolt-db/firebolt-go-sdk/actions/runs/3717140023/jobs/6304218056
I disagree on it being only dev/stg though. Those creds are for admin role with ability to create arbitrary number of engines, which can cause a massive AWS bill really quickly. It also can be used to overload the services effectively causing an outage.

Reply:
I've run several other tests trying to output the secrets and looks like they're always obscured so I think we might be reasonably safe here.